lib/graphql/static_validation/literal_validator.rb

```diff
@@ -6,10 +6,10 @@ def validate(ast_value, type)
     elsif type.kind.list? && ast_value.is_a?(Array)
       item_type = type.of_type
       ast_value.all? { |val| validate(val, item_type) }
-    elsif type.kind.scalar?
-      !type.coerce_input(ast_value).nil?
+    elsif type.kind.scalar? && !ast_value.is_a?(GraphQL::Language::Nodes::AbstractNode) && !ast_value.is_a?(Array)
```

Comment:
> yeah, seems like this should be the business of GraphQL::StaticValidation::ArgumentLiteralsAreCompatible.

ArgumentLiteralsAreCompatible mostly just uses this method to do the actual validation, so this is a part of its business.

> I think we should move that check here:
> 
> https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/static_validation/literal_validator.rb#L9

That is this line, which are commenting on.